lib/core/security/tokenRepository.js

```diff
@@ -189,9 +189,16 @@ class TokenRepository extends Repository {
 
     const parsedExpiresIn = parseTimespan(expiresIn);
 
+    let maxTTL;
+    if (type === 'apiKey') {
+      maxTTL = global.kuzzle.config.security.apiKey.maxTTL;
+    } else {
+      maxTTL = global.kuzzle.config.security.jwt.maxTTL;
+    }
+
     if ( ! bypassMaxTTL
-      && global.kuzzle.config.security.jwt.maxTTL > 0
-      && ( parsedExpiresIn > global.kuzzle.config.security.jwt.maxTTL
+      && maxTTL > 0
```

Comment:
Hm, edge case: setting this configuration `0` will act as if it was set to `-1`. Is this intended?
What behavior should we define if maxTTL is 0?

Proposal: it makes no sense having a TTL of a few milliseconds, so a lower bound value might be necessary. Something like 30s or 60s for instance, since anything below that is a bit too short to be really usable. So if maxTTL is set to anything between 0 and that value, it would be defaulted to that lower bound. 

wdyt?

Reply:
Your solution sounds good to me but I don't really know if we should decide about a minimum value since I don't know the level of rigidity we decided for other configuration properties
I think we need to handle that case anyway because -1 should be the only value that could disable the use of maxTTL
We also could just handle 0 as any other maxTTL value with `maxTTL >  -1` and let this to the user

Reply:
well, that's another way to see it: no restriction, let kuzzle act exactly as told, and if users want to put unusable configuration, then let them

I'm ok with that.

Reply:
Any other opinion ? @Shiranuit @Aschen

Reply:
No restriction makes sense, well documented is mandatory tho. :+1:

Reply:
I'm ok with both of your solutions, but maybe we should try to keep the same principle across Kuzzle, I mean, if we choose to not babysit the users and let them configure Kuzzle the way they intended too, even if this make no sense, then we should apply this principle all the time if possible to be coherent.